apps/browser/lib/hooks/useEvoService.ts

```diff
@@ -0,0 +1,184 @@
+import { useAddMessages } from "@/lib/mutations/useAddMessages";
+import { useChats } from "@/lib/queries/useChats";
+import { useAddVariable } from "@/lib/mutations/useAddVariable";
+import { useUpdateUserFiles } from "@/lib/hooks/useUpdateUserFiles";
+import { ChatLog } from "@/components/Chat";
+import { EvoThreadCallbacks, EvoThreadConfig } from "@/lib/services/evo/EvoThread";
+import { v4 as uuid } from "uuid";
+import { useAtom } from "jotai";
+import { useState, useEffect, useRef } from "react";
+import { Workspace, InMemoryWorkspace } from "@evo-ninja/agent-utils";
+import { ChatLogType, ChatMessage } from "@evo-ninja/agents";
+
+export const useEvoService = (
+  chatId: string | "<anon>" | undefined,
+  isAuthenticated: boolean,
+  onCreateChat: (chatId: string) => void
+): {
+  logs: ChatLog[] | undefined;
+  isStarting: boolean;
+  isRunning: boolean;
+  handleStart: (goal: string) => void;
+} => {
+  // Globals
+  const [evoService] = useAtom(evoServiceAtom);
+  const [allowTelemetry] = useAtom(allowTelemetryAtom);
+  const [openAiApiKey] = useAtom(localOpenAiApiKeyAtom);
+  const [, setUserWorkspace] = useAtom(userWorkspaceAtom);
+  const [, setCapReached] = useAtom(capReachedAtom);
+  const [, setAccountModalOpen] = useAtom(showAccountModalAtom);
+  const [, setError] = useAtom(errorAtom);
+
+  // State
+  const [isStarting, setIsStarting] = useState<boolean>(false);
+  const [isRunning, setIsRunning] = useState<boolean>(false);
+  const [chatLog, setChatLog] = useState<ChatLog[]>([]);
+
+  // Mutations
+  const { mutateAsync: createChat } = useCreateChat();
+  const { mutateAsync: addChatLog } = useAddChatLog();
+  const { mutateAsync: addMessages } = useAddMessages();
+  const { mutateAsync: addVariable } = useAddVariable();
+
+  // Queries
+  const { refetch: fetchChats } = useChats();
+
+  // Helpers
+  const updateUserFiles = useUpdateUserFiles();
+
+  const handleChatIdChange = (chatId: string | undefined) => {
+    const currentThread = evoService.current;
+
+    if (currentThread && currentThread.chatId === chatId) {
+      return;
+    }
+
+    evoService.disconnect();
+
+    const config: EvoThreadConfig = {
+      chatId: chatId || "<anon>",
+      loadChatLog,
+      loadWorkspace,
+      onChatLogAdded: handleChatLogAdded,
+      onMessagesAdded: handleMessagesAdded,
+      onVariableSet: handleVariableSet
+    };
+    const callbacks: EvoThreadCallbacks = {
+      setIsRunning,
+      setChatLog,
+      setWorkspace,
+      onGoalCapReached: () => {
+        setCapReached(true);
+        setAccountModalOpen(true);
+      },
+      onError: (error: string) => {
+        console.error(error);
+        setError("Failed to start Evo.");
+      },
+    };
+    evoService.connect(config, callbacks);
+  }
+
+  const loadChatLog = async (chatId: string) => {
+    if (chatId === "<anon>") {
+      return [];
+    }
+
+    const { data: chats } = await fetchChats();
```

Comment:
Shouldn't there be some sort of error handling here in case this promise fails?

Reply:
See the `EvoThread.load()` method, if an error is thrown it simply uses an empty array. Should we also have a `setError(...)` here?

Reply:
I think so